src/modules/csm/csm.py

```diff
@@ -258,7 +273,16 @@ def calculate_distribution(
 
             logs.append(log)
 
-        return total_distributed, total_rewards, logs
+        if total_distributed != sum(total_rewards.values()):
+            raise InconsistentData(f"Invalid distribution: {sum(total_rewards.values())=} != {total_distributed=}")
+
+        for no_id, rewards in last_report.rewards:
+            total_rewards[no_id] += rewards
```

Comment:
```suggestion
        for no_id, last_report_rewards in last_report.rewards:
            total_rewards[no_id] += last_report_rewards
```
Confused me a bit because of the same cycle as above.

Can we consider to have some func to merge the values? With validation. And reuse it here and in `frames` cycle